plugins/database/influxdb/influxdb.go

```diff
@@ -102,13 +102,17 @@ func (i *Influxdb) NewUser(ctx context.Context, req dbplugin.NewUserRequest) (re
 				"username": username,
 				"password": req.Password,
 			}
-			q := influx.NewQuery(dbutil.QueryHelper(query, m), "", "")
-			response, err := cli.Query(q)
-			if err != nil {
+			qry := influx.NewQuery(dbutil.QueryHelper(query, m), "", "")
+			response, err := cli.Query(qry)
+			// err can be nil with response.Error() being not nil, so both need to be handled
+			merr := multierror.Append(err, response.Error())
+			if merr.ErrorOrNil() != nil {
+				// Attempt rollback only when the response has an error
 				if response != nil && response.Error() != nil {
 					attemptRollback(cli, username, rollbackIFQL)
 				}
-				return dbplugin.NewUserResponse{}, fmt.Errorf("failed to run query in InfluxDB: %w", err)
+
+				return dbplugin.NewUserResponse{}, fmt.Errorf("failed to run query in InfluxDB: %w", merr.ErrorOrNil())
```

Comment:
Since this is already wrapped by a nil check, it should be safe to call `merr` right?

Reply:
Technically yes, though either works